internal/mock_matryer.templ

```diff
@@ -15,6 +15,15 @@ package {{.PkgName}}
     {{- $_ := .Registry.AddImport "sync" "sync" }}
 {{- end }}
 
+{{- if ne .PkgName .Registry.SrcPkg.Name }}
```

Comment:
Is this correct in the general case? Technically, the mocks being placed outside of the source package path can still have the same package name, so I think this logic would incorrectly determine them to be in the same package in that case.

Reply:
Valid point. I changed this to `{{- if ne $.SrcPkgQualifier "" }}`. It seems like this value is set only when not in package. So we should only consider doing this if we are not in the source package. I have also added some tests that checks the three different cases -generating within package, generating outside of package with different name, and outside package with the same name as the source package.